storage/pyproject.toml

```diff
@@ -26,6 +26,7 @@ typing-extensions = { version = "^4.6.3", python = "<3.8" }
 [tool.poetry.group.dev.dependencies]
 robocorp-devutils = { path = "../devutils/", develop = true }
 types-requests = "^2.30.0.0"
+black = { version = "^24.3.0" }
```

Comment:
I don't think we should pin **black** here, since we're not using it as a direct dependency, but as an [indirect](https://github.com/robocorp/lazydocs/blob/main/setup.py#L61) one through **lazydocs** which is coming from the _devutils/pyproject.toml_ that get's "inherited" in all our packages through this pin: `robocorp-devutils = { path = "../devutils/", develop = true }`

And I would pin it consistently like: `black = "^24.3.0"`

Then ensure I ran `inv install -u` in all our packages to make sure the poetry.lock file builds and maybe even run a `inv docs && inv check` (after an `inv devinstall`) to ensure that these commands relying on **lazydocs** are still working (which most probably are).

Reply:
Humm, lazydocs is our own fork at this point. We can remove the 'black' dep from it (we shouldn't have a black dependency at all anywhere we control).

Reply:
One thing I don't know: it'd be nice if we wouldn't have the `dev` deps of these projects at all in our lock files. Do we know if this is possible with poetry?

Reply:
Maybe just remove the black/lock files from this PR as that'd be a separate issue to be handled in a different PR and merge the other parts (which are unrelated to the black issue).

Reply:
I think we should keep the scope of this PR and fix the raised CVEs.
We can do improvements afterwards.

Reply:
Well, removing black would fix the CVE ;) 

But if it's trickier then we can go with the workaround until we can find out how to actually remove the dep.

Reply:
@fabioz Yes, there's an option to Poetry install certain groups with `--only main` or `--without dev`, thus not bringing in dev deps in the committed _*.lock_ files ([documentation](https://python-poetry.org/docs/cli/#:~:text=If%20you%20want%20to%20exclude,use%20the%20%2D%2Dwithout%20option.&text=The%20%2D%2Dno%2Ddev%20option,with%20the%20%2D%2Dwith%20option.)). But we'll need to apply something like this directly into the _devutils/pyproject.toml_ source when installing the **lazydocs** from Git source:
```toml
lazydocs = { git = "https://github.com/robocorp/lazydocs.git" }
```

The odd thing is that the **black** stemming from here is an extra, meaning that this will get installed only if we install like `lazydocs[dev]`, so in case of the `git` pin above, maybe there's a bug with Poetry or we have to pin by version (after a release of **lazydocs**) in order to not end up with its dev deps.

Even like that, **black** is brought by to other sub-deps:
- **linkify-it-py** -> dev: **black**
- **typer** -> test: **black** `>=22.3.0,<23.0.0`

So I'm not sure how the **black** removal from **lazydocs** will help here at all.